orca/server_metrics_test.go

```diff
@@ -126,3 +134,42 @@ func (s) TestServerMetrics_Setters_Range(t *testing.T) {
 		t.Fatalf("unexpected server metrics: -got +want: %v", d)
 	}
 }
+
+func (s) TestServerMetrics_Merge(t *testing.T) {
+	sm1 := &ServerMetrics{
+		CPUUtilization: 0.1,
+		MemUtilization: 0.2,
+		AppUtilization: 0.3,
+		QPS:            -1,
+		EPS:            0,
+		Utilization:    map[string]float64{"x": 0.6},
+		NamedMetrics:   map[string]float64{"y": 0.2},
+		RequestCost:    map[string]float64{"a": 0.1},
+	}
+
+	sm2 := &ServerMetrics{
+		CPUUtilization: -1,
+		AppUtilization: 0,
```

Comment:
Unfortunate that MemUtilization's zero value = set valid metric which overwrites, but can't do anything about that.

Reply:
Yeah the only option would be to use pointers and I don't really want to do that either.  We basically control everything around this struct and the values of the fields, though, so it's pretty safe.